logstash-core/src/main/java/org/logstash/ackedqueue/io/FileCheckpointIO.java

```diff
@@ -64,10 +63,12 @@ public Checkpoint write(String fileName, int pageNum, int firstUnackedPageNum, l
 
     @Override
     public void write(String fileName, Checkpoint checkpoint) throws IOException {
-        Path path = Paths.get(dirPath, fileName);
-        final byte[] buffer = new byte[BUFFER_SIZE];
         write(checkpoint, buffer);
-        Files.write(path, buffer, WRITE_OPTIONS);
+        buffer.flip();
+        try (FileOutputStream out = new FileOutputStream(Paths.get(dirPath, fileName).toFile())) {
```

Comment:
My thought on `assert` is that we are relying on a property of an object that cannot be checked by the compiler, which kind of stinks. 

Having an `assert` makes our tests fail if we miss something in code review (we miss things frequently, and that's ok, we are human). Similar to why I like tests because we cannot catch everything in code review (imperfect humans with finite time and attention available).

I am OK without an assert given the comment is a good hint.